src/dodal/beamlines/__init__.py

```diff
@@ -0,0 +1,11 @@
+ALL_BEAMLINES: set[str] = {
+    "i03",
+    "i04",
+    "i04_1",
+    "i20_1",
+    "i22",
+    "i23",
+    "i24",
+    "p38",
+    "p45",
+}
```

Comment:
This list of all beamline modules was previously maintained in the tests directory. It has been moved here so the CLI can reuse it. 

I'm not sure how much I like maintaining a list of all beamline modules as it seems prone to error. The advantage is that typing `dodal connect --help` will include a list of beamlines you can connect to.

The alternatives are some fancy auto-detection of beamline modules, perhaps by finding all modules in `dodal.beamlines` that match a pattern or simply parsing any string the user enters and printing an error message if it's not a beamline. 

I dislike the first because baking in beamline naming conventions has tended to lead to trouble in the past.

Reply:
We could move `_device_helpers.py` etc. into some other location and just have a convention that only beamline modules live in `beamlines`. Then we can just auto-detect based on all the module names in that folder?

Reply:
I felt that was worth its own PR: https://github.com/DiamondLightSource/dodal/pull/544